src/main/java/com/hivemq/client/internal/mqtt/handler/ssl/SslUtil.java

```diff
@@ -74,7 +60,17 @@ public static void initChannel(final @NotNull Channel channel, final @NotNull Mq
 
         sslContextBuilder.ciphers(sslConfig.getRawCipherSuites(), SupportedCipherSuiteFilter.INSTANCE);
 
-        return sslContextBuilder.build();
+        return new DelegatingSslContext(sslContextBuilder.build()) {
+            @Override
+            protected void initEngine(@NotNull final SSLEngine engine) {
+                engine.setUseClientMode(true);
```

Comment:
`engine.setUseClientMode(true)` can be removed, it was unnecessary.

Reply:
Removed